README.md

```diff
@@ -1,3 +1,192 @@
+
+`component`: SVG element to be rendered as the Node in the Network. `component` is a [render prop](https://reactjs.org/docs/render-props.html) recieves all of the props passed to `Node` and returns a <svg> react component. The svg will be rendered as the `image` of the node. Helpful tip, you can pass additional props to `node` that can be used in your render prop to dynamically change your image.
+
+```js
+const CustomIcon = ({ icon, color = '#5596ed' }) => {
+  const viewBox = 36;
+  const iconSize = 20;
+  const pad = (viewBox - iconSize) / 2;
+  const center = viewBox / 2;
+
+  return (
+    <svg
+      xmlns="http://www.w3.org/2000/svg"
+      viewBox={`0 0 ${viewBox} ${viewBox}`}
+    >
+      <g>
+        <circle cx={center} cy={center} r={16} fill={color} />
+        <g transform={`translate(${pad}, ${pad})`}>
+          {React.createElement(icon, { color: 'white', size: iconSize })}
+        </g>
+      </g>
+    </svg>
+  );
+};
+
+// Use your render prop and custom props passed to `Node` to
+// Generate differnt icons per node
+const MyNetwork = () => (
+  <Network>
+    <Node
+      id="vader"
+      label="Darth Vader"
+      component={CustomIcon}
+      color="black"
+      icon={Sith}
+    />
+    <Node
+      id="luke"
+      label="Luke Skywalker"
+      component={CustomIcon}
+      color="white"
+      icon={Jedi}
+    />
+    <Node
+      id="leia"
+      label="Leia Organa"
+      component={CustomIcon}
+      color="gold"
+      icon={Princess}
+    />
+    <Edge id="1" from="vadar" to="luke" />
+    <Edge id="2" from="vadar" to="leia" />
+  </Network>
+);
+```
+
+**NB:** This is simply syntactic sugar over [using HTML in Nodes](https://github.com/almende/vis/blob/master/examples/network/nodeStyles/HTMLInNodes.html) and will end up being a data+image. You will not recieve dom events through this component, use `Network`'s event system or `decorator`s instead.
+
+`decorator`: A react element positioned in the center of rendered `Node`. A decorator is similar to `component` in that it's a [render prop](https://reactjs.org/docs/render-props.html) that recieves the props of a the node, and returns a react element. This element however doesn't need to be an svg.
+
+```js
+const Decorator = props => {
+  return (
+    <button
+      style={decoratorStyles}
+      onClick={() => console.log(`You clicked ${props.label}`}
+    >
+      Click Me
+    </button>
+  );
+};
+
+// Use your render prop and custom props passed to `Node` to
+// cause differnet actions in your Decorator
+const MyNetwork = () => (
+  <Network>
+    <Node id='vadar' label='Darth Vadar' decorator={Decorator} />
+    <Node id='luke' label='Luke Skywalker' decorator={Decorator} />
+    <Node id='leia' label='Leia Organa' decorator={Decorator} />
+    <Edge id='1' from='vadar' to='luke' />
+    <Edge id='1' from='vadar' to='leia' />
```

Comment:
Might as well call it out as a global search/replace for `Vadar` => `Vader` again. ;]